test/Oscoin/Test/Storage/Block/Cache.hs

```diff
@@ -0,0 +1,50 @@
+module Oscoin.Test.Storage.Block.Cache
+    ( tests
+    ) where
+
+import           Oscoin.Prelude
+
+import           Oscoin.Crypto.Blockchain.Block hiding (genesisBlock)
+
+import           Oscoin.Storage.Block.Cache
+                 (consBlock, newBlockCache, viewBlocks)
+import           Oscoin.Test.Crypto
+import           Oscoin.Test.Crypto.Blockchain.Block.Arbitrary ()
+
+import           Data.ByteArray.Orphans ()
+
+import           Test.QuickCheck.Monadic
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+
+tests :: forall c. Dict (IsCrypto c) -> [TestTree]
+tests Dict =
+    [ testGroup "Storage.Block.Cache"
+        [ testProperty "works as a sequence"    (worksSequenceProp @c Dict)
+        , testProperty "works as a ring buffer" (worksRingBufferProp @c Dict)
+        ]
+    ]
+
+{------------------------------------------------------------------------------
+  The tests proper
+------------------------------------------------------------------------------}
+
+worksSequenceProp :: forall c. Dict (IsCrypto c) -> Property
+worksSequenceProp Dict = monadicIO $ do
+    cache <- liftIO (newBlockCache 10)
+    forAllM (arbitrary `suchThat` ((< 10) . length)) $ \(blks :: [Block c Text Text]) -> do
+        storedBlocks <- liftIO $ do
+            forM_ (reverse blks) (consBlock cache)
+            viewBlocks cache
+        assert $ storedBlocks == blks
+
+worksRingBufferProp :: forall c. Dict (IsCrypto c) -> Property
+worksRingBufferProp Dict = monadicIO $ do
+    cache <- liftIO (newBlockCache 10)
+    forAllM (arbitrary `suchThat` ((> 10) . length)) $ \(blks :: [Block c Text Text]) -> do
+        storedBlocks <- liftIO $ do
+            forM_ (reverse blks) (consBlock cache)
+            viewBlocks cache
+        assert $ head storedBlocks == head blks
+        assert $ storedBlocks /= blks
```

Comment:
We could drop the two assertions and replace them with
```suggestion
        assert $ storedBlocks == take 10 blks
```

Reply:
3d20d7a